### PR TITLE
First unit test and Github action

### PR DESCRIPTION
**Pull request by simonbaird**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (joejstuart) [inline Makefile:1]: Do we run `make test` from the root directory or from the directory where the rego files are?

**Comment by simonbaird** [comment]
I added some improvements including a Github pipeline, see here:

* https://github.com/simonbaird/ec-policies/actions/runs/2073384955

File: Makefile [hunk 2]
```diff
@@ -2,11 +2,27 @@
...
+# Rewrite all rego files with the preferred format
+# Use before you commit
+fmt:
+	@opa fmt . --write
+
+# Return non-zero exit code if formatting is needed
+# Used in CI
+fmt-check:
+	@opa fmt . --list | xargs -r -n1 echo 'Incorrect formatting found in'
+	@opa fmt . --list --fail >/dev/null 2>&1
```
Comment (simonbaird) [inline Makefile:21]: Slightly inefficient to run it twice but should be ok..

**Comment by simonbaird** [comment]
I might merge it soon if there's no objections.

**Comment by zregvart** [comment]
image

File: .github/workflows/pre-merge-ci.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
+name: Pre-merge CI
+
+on:
```
Comment (lcarva) [inline .github/workflows/pre-merge-ci.yaml:4]: I think we want to run this again on the on `push` to verify that after a PR is merged, no issues are introduced.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+# For convenience. If this passes then it should pass in GitHub
+ci: fmt-check test
+
+OPA_VER=v0.39.0
+OPA_URL=https://openpolicyagent.org/downloads/$(OPA_VER)/opa_linux_amd64_static
+OPA_DEST=/usr/bin/opa
+
+install-opa:
+	sudo curl -s -L -o $(OPA_DEST) $(OPA_URL)
```
Comment (lcarva) [inline Makefile:31]: It would be nice to do a checksum verification.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	sudo chmod 755 $(OPA_DEST)
+
+.PHONY: help test install-opa
```
[the remark this answers is not in the document]
Comment (simonbaird) [reply Makefile:34]: Thanks, will fix in upcoming PR.

File: Makefile (hunk 4 above)
Comment (simonbaird) [reply Makefile:31]: Agreed. https://github.com/hacbs-contract/ec-policies/pull/3

File: .github/workflows/pre-merge-ci.yaml (hunk 3 above)
Comment (simonbaird) [reply .github/workflows/pre-merge-ci.yaml:4]: Good point. I did a sneaky rewrite and force push.

File: Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (simonbaird) [reply Makefile:34]: Thanks. Done with a rewrite and force push.

**Pull request merged**